app/templates/talk/page.tsx

```diff
@@ -5,17 +5,29 @@ import {useInputChange} from '../../../src/app/hooks/onInputChange';
 import {Form, Input} from '../../../src/app/forms/input';
 import {Code} from '../../../src/app/Code';
 import {Talk} from '../../../remotion/compositions/templates/talk/Talk';
+import {useSearchParams} from 'next/navigation';
+import {CopyUrlButton} from '../../../src/app/CopyUrlButton';
+import {encodeObjectValues} from '../../../src/app/utils/encodeObjectValues';
 
 export default function TalkPage() {
-	const [talkTitle, setTalkTitle] = useInputChange<string>('Example');
+	const searchParams = useSearchParams();
+	const [talkTitle, setTalkTitle] = useInputChange<string>(
+		searchParams.get('talkTitle') || 'Example'
+	);
 	const [speakerPicture, setSpeakerPicture] = useInputChange<
 		string | undefined
-	>(undefined);
-	const [speakersNames, setSpeakersNames] = useInputChange<string>('John Doe');
-	const [eventLogo, setEventLogo] = useInputChange<string>('');
-	const [titleSize, setTitleSize] = useInputChange<string>('50');
+	>(searchParams.get('speakerPicture') || undefined);
+	const [speakersNames, setSpeakersNames] = useInputChange<string>(
+		searchParams.get('speakersNames') || 'John Doe'
+	);
+	const [eventLogo, setEventLogo] = useInputChange<string>(
+		searchParams.get('eventLogo') || ''
+	);
+	const [titleSize, setTitleSize] = useInputChange<string>(
+		searchParams.get('titleSize') || '50'
+	);
 	const [backgroundImg, setBackgroundImg] = useInputChange<string | undefined>(
-		undefined
+		searchParams.get('backgroundImg') || undefined
 	);
```

Comment:
`useInputChange` could take a new param which is the name of the queryparams ?

Reply:
To move searchParams.get() on useInputChange() ?

Reply:
okay got it